### DISCUSSION TITLE
Green Grass Certificates - can it be used for https?

### DISCUSSION THREAD

**Discussion by chandar1** [aws/aws-iot-device-sdk-cpp-v2#727] [Q&A] [answered]
Hi,

I have installed Green Grass on my device, and it has root CA, private key and thing certificate to connect to IoT core, like for MQTT connections.

I also have a https server running on the device that be accessed from any client, and can I re-use the Green Grass certificates to host my https server. Any implications?
Idea is to use single certificate identity for a device....mqtt/tls and https .

I did try to use the certs for https, though on the client browser getting error,
```
Your connection isn't private,
Attackers might be trying to steal your information from <ip address> (for example, passwords, messages, or credit cards).
NET::ERR_CERT_AUTHORITY_INVALID 
```
 How to fix this pl? 

Any help is appreciated!

**Comment by bretambrose** [comment, marked as answer]
This doesn't seem like a question within the scope of the IoT SDK.  I am not a security expert, but re-using one service's X509 materials for a completely independent service's authentication does not seem like a good idea.
